packages/admin-portal/src/sagas/customers/customers-list.ts

```diff
@@ -0,0 +1,48 @@
+import { put, fork, takeLatest, all, call } from '@redux-saga/core/effects'
+import { fetchCustomersListFailed, fetchCustomersListSuccess, FetchCustomersListQueryParams } from '@/actions/customers'
+import { fetchCustomersList as fetchCustomersListAPI } from '@/services/customers'
+
+import { notification } from '@reapit/elements'
+import { Action } from '@/types/core'
+import ActionTypes from '@/constants/action-types'
+import { CUSTOMERS_PER_PAGE } from '@/constants/paginator'
+
+export const DEFAULT_PAGE = 1
+
+export const fetchCustomersListHandler = function*({ data: { queryString } }) {
+  try {
+    const queryParams = new URLSearchParams(queryString)
+    const pageNumber = parseInt(queryParams.get('page') as string)
+    const page = pageNumber < 1 || isNaN(pageNumber) ? DEFAULT_PAGE : pageNumber
+    const name = queryParams.getAll('name')
+    const agencyCloudId = queryParams.getAll('agencyCloudId')
+
+    const response = yield call(fetchCustomersListAPI, {
+      pageSize: CUSTOMERS_PER_PAGE,
+      pageNumber: page,
+      name,
+      agencyCloudId,
+    })
+    if (response) {
+      yield put(fetchCustomersListSuccess(response))
+      return
+    }
+    throw new Error('Can not fetch customers list !')
```

Comment:
This error string used many times (one from here, one in the testcase), maybe you want to extract this into a constant